doc/reference/vm_live_migration_internals.md

```diff
@@ -0,0 +1,96 @@
+---
+discourse: 50734
+---
+
+(vm-live-migration-internals)=
+# LXD VM live migration
+
+LXD supports {ref}`live-migration-vms` of virtual machines by streaming instance state from a source {abbr}`QEMU (Quick Emulator)` to a target QEMU. VM live migration is supported for all storage pool types.
+
+API extension: `migration_vm_live`
+
+## Conceptual process
+
+The live migration workflow varies depending on the type of storage pool used. The two key scenarios are non-shared storage and shared storage within a cluster (e.g., Ceph). If live state transfer is not supported by a target, a stateful stop is performed prior to migration.
+
+### Live migration for non-shared storage
+
+This process leverages the QEMU built-in Network Block Device (NBD) client-server mechanism to transfer the virtual machine's disk and memory state. Below is an overview of the steps:
+
+1. Set {abbr}`QMP (QEMU Machine Protocol)` capabilities.
+2. Temporary write buffer: Create a temporary {abbr}`QCOW2 (QEMU Copy on Write)` file to store writes that occur in the guest during the initial transfer of the root disk.
+3. Snapshot root disk.
+4. Database record creation: Create the target instance and volume database records.
+5. Storage driver transfer.
+6. Start target QEMU: Launch the QEMU process on the target.
+7. Add NBD node: Attach the NBD target disk as a block node on the source. 
+8. Block device mirror: Perform a block device mirror of the snapshot to the target NBD server.
+9. Pause source VM: Pause the source VM once the block device mirror enters `ready` state.
+10. State transfer: Transfer the memory state over WebSocket.
+11. Enter `pre-switchover` state.
+12. Complete migration: Continue state migration until `completed` state reached.
```

Comment:
I'd add the target VM is then resumed from the received and loaded state.
The source VM is stopped and its storage discarded.